### PR TITLE
Remove meshcatalog, getSmiPoliciesHandler endpoint, and getOSMConfigHandler endpoint from debugger

### PR DESCRIPTION
**Pull request by nshankar13**
Signed-off-by: nshankar13 <nshankar@microsoft.com>

**Description**: Remove the meshcatalog from the debug config. This entails calling the meshcatalog methods from the compute Client now. Also, ListServiceAccountsFromTrafficTargets is moved from the catalog pkg to compute client. No functional changes. 

Update: After an offline discussion, it was decided that the getSmiPoliciesHandler and getOSMConfigHandler endpoints can be removed altogether. 

Resolves: https://github.com/openservicemesh/osm/issues/5219

**Testing done**:

**Affected area**:
| Functional Area            |     |
| -------------------------- | --- |
| New Functionality          | [ ] |
| CI System                  | [ ] |
| CLI Tool                   | [ ] |
| Certificate Management     | [ ] |
| Control Plane              | [ ] |
| Demo                       | [ ] |
| Documentation              | [ ] |
| Egress                     | [ ] |
| Ingress                    | [ ] |
| Install                    | [ ] |
| Networking                 | [ ] |
| Observability              | [ ] |
| Performance                | [ ] |
| SMI Policy                 | [ ] |
| Security                   | [ ] |
| Sidecar Injection          | [ ] |
| Tests                      | [ ] |
| Upgrade                    | [ ] |
| Other                      | [ ] |

Please answer the following questions with yes/no.

1. Does this change contain code from or inspired by another project?
    -   Did you notify the maintainers and provide attribution?

2. Is this a breaking change?

3. Has documentation corresponding to this change been updated in the [osm-docs](https://github.com/openservicemesh/osm-docs) repo (if applicable)?

### CODE REVIEW

File: pkg/debugger/policy_test.go [hunk 1]
```diff
@@ -4,49 +4,46 @@ import (
 	"net/http/httptest"
 	"testing"
 
-	"github.com/golang/mock/gomock"
+	gomock "github.com/golang/mock/gomock"
+	"github.com/openservicemesh/osm/pkg/compute/kube"
+	"github.com/openservicemesh/osm/pkg/k8s"
```
Comment (jsturtevant) [inline pkg/debugger/policy_test.go:9]: this linter error is annoying 😃 fyi you can configure your IDE to do this automatically which has solved having to update this everytime

**Comment by jsturtevant** [comment]
lgtm

File: pkg/debugger/policy_test.go (hunk 1 above)
Comment (nshankar13) [reply pkg/debugger/policy_test.go:9]: Actually probably going to get rid of this file altogether due to upcoming change I'm about to make, but will keep this in mind!

File: pkg/debugger/namespace_test.go [hunk 2]
```diff
@@ -4,30 +4,46 @@ import (
...
-
-	mock := catalog.NewMockMeshCataloger(gomock.NewController(t))
-
-	ds := DebugConfig{
-		meshCatalog: mock,
-	}
-	monitoredNamespacesHandler := ds.getMonitoredNamespacesHandler()
+	mockCtrl := gomock.NewController(t)
+	mockK8s := k8s.NewMockController(mockCtrl)
+	computeClient := kube.NewClient(mockK8s)
```
Comment (steeling) [inline pkg/debugger/namespace_test.go:23]: you can just mock out the compute client itself

File: pkg/debugger/policy.go [hunk 3]
```diff
@@ -1,25 +1,12 @@
...
 )
 
-type policies struct {
-	TrafficSplits   []*split.TrafficSplit        `json:"traffic_splits"`
-	ServiceAccounts []identity.K8sServiceAccount `json:"service_accounts"`
-	RouteGroups     []*spec.HTTPRouteGroup       `json:"route_groups"`
-	TrafficTargets  []*access.TrafficTarget      `json:"traffic_targets"`
-}
-
 func (ds DebugConfig) getOSMConfigHandler() http.Handler {
```
Comment (steeling) [inline pkg/debugger/policy.go:10]: i don't see any reason to keep `getOSMConfigHandler` around either

**Review by steeling — APPROVED** [review]
Can you change the description and the title to reflect the new changes?

**Comment by jsturtevant** [comment]
>Update: After an offline discussion, it was decided that the getSmiPoliciesHandler and getOSMConfigHandler endpoints can be removed altogether.

For posterity, what was the reasoning behind removing these?  Is there any tooling like the bug check that uses those endpoints?

**Comment by nshankar13** [comment]
@jsturtevant we decided to get rid of these endpoints because a user can use kubectl to get the same info. I searched through the repo and these endpoints don't get invoked anywhere else.

**Review by jsturtevant — APPROVED** [review]

**Review by keithmattix — APPROVED** [review]

**Pull request merged**